lib/resty/session/file/utils.lua

```diff
@@ -35,25 +36,30 @@ local function get_path(storage, name, key)
 end
 
 
-local function get_path_meta(storage, key)
-  local path   = storage.path
-  local prefix = storage.prefix
-  local suffix = storage.suffix
+local function meta_get_key(audience, subject)
+  return fmt("%s:%s", audience, subject)
+end
 
-  if prefix and suffix then
-    return fmt("%s%s_%s.%s", path, prefix, key, suffix)
-  elseif prefix then
-    return fmt("%s%s_%s", path, prefix, key)
-  elseif suffix then
-    return fmt("%s%s.%s", path, key, suffix)
-  else
-    return fmt("%s%s", path, key)
+
+local function validate_file_name(storage, name, filename)
+  if filename == "." or filename == ".." then
+    return false
   end
+
+  local prefix = storage.prefix or ""
+  local suffix = storage.suffix or ""
+
+  local pre_name = (#prefix > 0 and prefix .. "_" .. name or name) .. "_"
+  local dot_suff = #suffix > 0 and "\\." .. suffix or ""
+
+  local pattern = fmt("^%s([\\w_-]{43}|.+:.+)%s$", pre_name, dot_suff)
+  return re_match(filename, pattern) and true or false
 end
```

Comment:
I was thinking something that just uses fast and simple stuff, I have not checked below code, but you get an idea:

```suggestion
local function validate_file_name(storage, name, filename)
  if filename == "." or filename == ".." then
    return false
  end

  local plen = 0
  local prefix = storage.prefix
  if prefix then
    plen = #prefix
    if byte(filename, plen + 1) ~= UNDERSCORE then
      return false
    end

    if plen > 0 and sub(filename, 1, plen) ~= prefix then
      return false
    end
  end

  local slen = 0
  local suffix = storage.suffix
  if suffix then
    slen = #suffix
    if byte(filename, -1 - slen) ~= DOT then
      return false
    end

    if slen > 0 and sub(filename, 1, -slen) ~= suffix then
      return false
    end
  end

  local nlen = #name
  local name_start = plen == 0 and 1 or plen + 2
  local name_end = name_start + nlen - 1

  if byte(filename, name_end + 1) ~= UNDERSCORE then
    return false
  end

  if sub(filename, name_start, name_end) ~= name then
    return false
  end

  local rest
  if slen == 0 then
    rest = sub(filename, name_end + 2)
  else
    rest = sub(filename, name_end + 2, -2 - slen)
  end
  
  local rlen = #rest
  if rlen < 3 then
    return false
  end
  
  if rlen ~= 43 then
    local colon_pos = find(rest, ":", 2, true)
    if not colon_pos or colon_pos == 43 then
      return false
    end
  end
  
  return true
end
```